trogon/trogon.py

```diff
@@ -217,24 +210,47 @@ class Trogon(App):
 
     def __init__(
         self,
-        cli: click.Group,
-        app_name: str | None = None,
-        command_name: str = "tui",
-        click_context: click.Context | None = None,
+        command_schemas: dict[CommandName, CommandSchema],
+        app_name: str | None,
+        app_version: str | None = None,
```

Comment:
`Trogon` now accepts the schemas directly, rather than inferring schemas from a given CLI. It is the responsibility of the various "introspectors" to do the reverse-engineering and provide schemas to `CommandBuilder`.